### PR TITLE
Logcollector integration tests T0: reconnect_time option

### LINKED ISSUE #1251: Logcollector integration tests T0: Check reconnect_time
As part of [ Logcollector integration tests: Tier 0 #1027 ](https://github.com/wazuh/wazuh-qa/issues/1027), we need to add a test to check if after disconnecting from Windows Event Channel and connecting again, the daemon continues reading events from it and reporting them.

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|Closes #1251 |

## Description

This PR adds logcollector tests for [reconnect_time](https://documentation-dev.wazuh.com/current/user-manual/reference/ossec-conf/localfile.html?highlight=only%20future%20event#reconnect_time) option. 

  - [x] **Test reconnect time**: Check that logcollector reconnect to Windows event log using `reconnect_time` value.
 
This test is valid for Windows agents.

Every case of `reconnect_time` greater than 30 seconds will be marked as xfail due to this issue: https://github.com/wazuh/wazuh/issues/8580

### Windows - Agent

reconnect_time_logcollector

### Test reconnect time 
test_reconnec_time_docu

### CODE REVIEW

File: tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    """Check if reconnect_time value works properly
+
+    Ensure correspond debug logs are generated when Windows event log service stop. Also, when event log service is
+    restarted, `wazuh-agent` should reconnect to it using reconnect_time value.
+    """
+
+    config = get_configuration['metadata']
+
+    if time_to_seconds(config['reconnect_time']) >=  timeout_callback_reconnect_time:
+        pytest.xfail("Expected fail: ")
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py:85]: Add more detail here explaining why the error is expected.

File: tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        before = str(datetime.now())
+        seconds_to_travel = time_to_seconds(config['reconnect_time']) / 2
+        TimeMachine.travel_to_future(timedelta(seconds=seconds_to_travel))
+        logger.debug(f"Changing the system clock from {before} to {datetime.now()}")
+
+    log_callback = logcollector.callback_reconnect_eventchannel(config['location'])
+
+    before = str(datetime.now())
+
+    if time_to_seconds(config['reconnect_time']) >=  timeout_callback_reconnect_time:
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py:116]:
```suggestion
    if time_to_seconds(config['reconnect_time']) >= timeout_callback_reconnect_time:
```

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 3]
```diff
@@ -199,3 +199,62 @@ def callback_running_command(log_format, command, prefix=monitoring.LOG_COLLECTO
...
+    Returns:
+        callable: callback to detect this event.
+    """
+    log_format_message = 'full command' if log_format == 'full_command' else 'command'
+    msg = fr"DEBUG: Running {log_format_message} '{command}'"
+    return monitoring.make_callback(pattern=msg, prefix=prefix, escape=escape)
+
+
+
+def callback_event_log_service_down(location, severity='WARNING'):
```
[the remark this answers is not in the document]
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/logcollector.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 4]
```diff
@@ -199,3 +199,62 @@ def callback_running_command(log_format, command, prefix=monitoring.LOG_COLLECTO
...
+        escape (bool): Flag to escape special characters in the pattern.
+
+    Returns:
+        callable: callback to detect this event.
+    """
+    log_format_message = f"{severity}: The eventlog service is down. Unable to collect logs from '{location}' channel."
+    print(f"{log_format_message}")
+    return monitoring.make_callback(pattern=log_format_message, prefix=monitoring.AGENT_DETECTOR_PREFIX)
+
+def callback_trying_to_reconnect(location, reconnect_time):
```
[the remark this answers is not in the document]
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/logcollector.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 5]
```diff
@@ -199,3 +199,62 @@ def callback_running_command(log_format, command, prefix=monitoring.LOG_COLLECTO
...
+        escape (bool): Flag to escape special characters in the pattern.
+
+    Returns:
+        callable: callback to detect this event.
+    """
+    log_format_message = f"DEBUG: Trying to reconnect {location} channel in {reconnect_time} seconds."
+    print(f"{log_format_message}")
+    return monitoring.make_callback(pattern=log_format_message, prefix=monitoring.AGENT_DETECTOR_PREFIX)
+
+def callback_reconnect_eventchannel(location):
```
[the remark this answers is not in the document]
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/logcollector.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 6]
```diff
@@ -199,3 +199,62 @@ def callback_running_command(log_format, command, prefix=monitoring.LOG_COLLECTO
...
+    """
+
+    """
```
[the remark this answers is not in the document]
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/logcollector.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: docs/tests/integration/test_logcollector/test_reconnect_time.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+## General info
+
+|Tier | Number of tests | Time spent |
+|:--:|:--:|:--:|
+| 0 | 9 | 1m4s |
+
+## Expected behavior
+
+- Fail if Wazuh agent does not reconnect to Windows event log at the time specified by `reconnect_time `
+- Fail if Wazuh agent does not generate a debug message when Windows event log is down
```
[the remark this answers is not in the document]
Comment (Rebits) [reply docs/tests/integration/test_logcollector/test_reconnect_time.md]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+from wazuh_testing import global_parameters, logger
+from wazuh_testing.tools.time import TimeMachine
+import wazuh_testing.logcollector as logcollector
+from wazuh_testing.tools.time import time_to_seconds
+import wazuh_testing.tools.services as services
+
+if sys.platform != 'win32':
+    pytestmark = [pytest.mark.skip, pytest.mark.tier(level=0)]
+else:
+    pytestmark = pytest.mark.tier(level=0)
```
[the remark this answers is not in the document]
Comment (Rebits) [reply tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py (hunk 1 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

File: tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py (hunk 2 above)
Comment (Rebits) [reply tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py]: Done 4d788e12e477bbb1f3655163791253a96b576e07

**Review by BraulioV — APPROVED** [review]

**Pull request merged**